File: src/shared/adios_method_stubs.f90
!=====================================================================
!
!          S p e c f e m 3 D  G l o b e  V e r s i o n  6 . 0
!          --------------------------------------------------
!
!          Main authors: Dimitri Komatitsch and Jeroen Tromp
!                        Princeton University, USA
!             and CNRS / INRIA / University of Pau, France
! (c) Princeton University and CNRS / INRIA / University of Pau
!                            August 2013
!
! This program is free software; you can redistribute it and/or modify
! it under the terms of the GNU General Public License as published by
! the Free Software Foundation; either version 2 of the License, or
! (at your option) any later version.
!
! This program is distributed in the hope that it will be useful,
! but WITHOUT ANY WARRANTY; without even the implied warranty of
! MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
! GNU General Public License for more details.
!
! You should have received a copy of the GNU General Public License along
! with this program; if not, write to the Free Software Foundation, Inc.,
! 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
!
!=====================================================================


! placeholders for non-adios compilation

  subroutine warn_no_adios()
    stop 'Trying to use ADIOS while not configured for it.'
  end subroutine warn_no_adios

! modules

!  module AVS_DX_global_mod
!    type avs_dx_global_t
!    end type avs_dx_global_t
!  end module AVS_DX_global_mod

!  module AVS_DX_global_faces_mod
!    type avs_dx_global_faces_t
!    end type avs_dx_global_faces_t
!  end module AVS_DX_global_faces_mod

!  module AVS_DX_global_chunks_mod
!    type avs_dx_global_chunks_t
!    end type avs_dx_global_chunks_t
!  end module AVS_DX_global_chunks_mod

!  module AVS_DX_surface_mod
!    type avs_dx_surface_t
!    end type avs_dx_surface_t
!  end module AVS_DX_surface_mod

! for both xmeshfem3D/xspecfem3D compilation

  subroutine adios_setup()
  call warn_no_adios()
  end subroutine

  subroutine adios_cleanup()
  end subroutine

! for xmeshfem3D compilation

  subroutine crm_save_mesh_files_adios()
  end subroutine

  subroutine get_absorb_adios()
  end subroutine

  subroutine save_arrays_solver_adios()
  end subroutine

  subroutine save_arrays_solver_meshfiles_adios()
  end subroutine

  subroutine save_arrays_solver_boundary_adios()
  end subroutine

  subroutine save_MPI_arrays_adios()
  end subroutine

  subroutine read_gll_model_adios
  end subroutine
  
!  subroutine prepare_AVS_DX_global_chunks_data_adios()
!  end subroutine
!
!  subroutine write_AVS_DX_global_chunks_data_adios()
!  end subroutine
!
!  subroutine free_AVS_DX_global_chunks_data_adios()
!  end subroutine
!
!  subroutine define_AVS_DX_global_data_adios()
!  end subroutine
!
!  subroutine prepare_AVS_DX_global_data_adios()
!  end subroutine
!
!  subroutine write_AVS_DX_global_data_adios()
!  end subroutine
!
!  subroutine free_AVS_DX_global_data_adios()
!  end subroutine
!
!  subroutine define_AVS_DX_global_faces_data_adios()
!  end subroutine
!
!  subroutine prepare_AVS_DX_global_faces_data_adios()
!  end subroutine
!
!  subroutine write_AVS_DX_global_faces_data_adios()
!  end subroutine
!
!  subroutine free_AVS_DX_global_faces_data_adios()
!  end subroutine
!
!  subroutine define_AVS_DX_surfaces_data_adios()
!  end subroutine
!
!  subroutine prepare_AVS_DX_surfaces_data_adios()
!  end subroutine
!
!  subroutine write_AVS_DX_surfaces_data_adios()
!  end subroutine
!
!  subroutine free_AVS_DX_surfaces_data_adios()
!  end subroutine
!

! for xspecfem3D compilation

  subroutine read_arrays_solver_adios()
  end subroutine

  subroutine read_attenuation_adios()
  end subroutine

  subroutine read_forward_arrays_adios()
  end subroutine

  subroutine read_intermediate_forward_arrays_adios()
  end subroutine

  subroutine read_mesh_databases_coupling_adios()
  end subroutine

  subroutine read_mesh_databases_mpi_cm_adios()
  end subroutine

  subroutine read_mesh_databases_mpi_ic_adios()
  end subroutine

  subroutine read_mesh_databases_mpi_oc_adios()
  end subroutine

  subroutine read_mesh_databases_stacey_adios()
  end subroutine

  subroutine save_forward_arrays_adios()
  end subroutine

  subroutine save_intermediate_forward_arrays_adios()
  end subroutine

!  subroutine define_common_forward_arrays_adios()
!  end subroutine
!
!  subroutine define_rotation_forward_arrays_adios()
!  end subroutine
!
!  subroutine define_attenuation_forward_arrays_adios()
!  end subroutine
!
!  subroutine write_common_forward_arrays_adios()
!  end subroutine
!
!  subroutine write_rotation_forward_arrays_adios()
!  end subroutine
!
!  subroutine write_attenuation_forward_arrays_adios()
!  end subroutine
!
!  subroutine write_1D_global_array_adios_dims()
!  end subroutine
!

  subroutine perform_write_adios_kernels()
  end subroutine

  subroutine define_kernel_adios_variables()
  end subroutine

  subroutine write_kernels_crust_mantle_adios()
  end subroutine

  subroutine write_kernels_outer_core_adios()
  end subroutine

  subroutine write_kernels_inner_core_adios()
  end subroutine

  subroutine write_kernels_boundary_kl_adios()
  end subroutine

  subroutine write_kernels_source_derivatives_adios()
  end subroutine

  subroutine write_kernels_hessian_adios()
  end subroutine

!
!  subroutine write_specfem_header_adios()
!  end subroutine

File: src/shared/adios_method_stubs.c
/*
!=====================================================================
!
!          S p e c f e m 3 D  G l o b e  V e r s i o n  7 . 0
!          --------------------------------------------------
!
!     Main historical authors: Dimitri Komatitsch and Jeroen Tromp
!                        Princeton University, USA
!                and CNRS / University of Marseille, France
!                 (there are currently many more authors!)
! (c) Princeton University and CNRS / University of Marseille, April 2014
!
! This program is free software; you can redistribute it and/or modify
! it under the terms of the GNU General Public License as published by
! the Free Software Foundation; either version 3 of the License, or
! (at your option) any later version.
!
! This program is distributed in the hope that it will be useful,
! but WITHOUT ANY WARRANTY; without even the implied warranty of
! MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
! GNU General Public License for more details.
!
! You should have received a copy of the GNU General Public License along
! with this program; if not, write to the Free Software Foundation, Inc.,
! 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
!
!=====================================================================
*/

#include <stdio.h>
#include <stdlib.h>

#include "config.h"

typedef float realw;


// for xmeshfem3D compilation

void FC_FUNC_(get_absorb_adios,GET_ABSORB_ADIOS)(int* iregion,
                                                 int* nimin, int* nimax, int* njmin, int* njmax, int* nkmin_xi, int* nkmin_eta,
                                                 int* NSPEC2DMAX_XMIN_XMAX, int* NSPEC2DMAX_YMIN_YMAX) {}

void FC_FUNC_(read_gll_model_adios,READ_GLL_MODEL_ADIOS)(void) {}

void FC_FUNC_(save_arrays_solver_adios,SAVE_ARRAYS_SOLVER_ADIOS)(int* idoubling, int* ibool,
                                                                 int* iregion_code,
                                                                 realw* xstore, realw* ystore, realw* zstore,
                                                                 int* NSPEC2DMAX_XMIN_XMAX, int* NSPEC2DMAX_YMIN_YMAX,
                                                                 int* NSPEC2D_TOP, int* NSPEC2D_BOTTOM) {}

void FC_FUNC_(save_arrays_solver_meshfiles_adios,SAVE_ARRAYS_SOLVER_MESHFILES_ADIOS)(void) {}

void FC_FUNC_(save_arrays_boundary_adios,SAVE_ARRAYS_BOUNDARY_ADIOS)(void) {}

void FC_FUNC_(save_mpi_arrays_adios,SAVE_MPI_ARRAYS_ADIOS)(void) {}

void FC_FUNC_(write_avs_dx_output_adios,WRITE_AVS_DX_OUTPUT_ADIOS)(int* npointot, int* iregion_code,
                                                                   int* num_ibool_AVS_DX, int* mask_ibool) {}

// for xspecfem3D compilation

void FC_FUNC_(read_arrays_solver_adios,READ_ARRAYS_SOLVER_ADIOS)(void) {}

void FC_FUNC_(read_attenuation_adios,READ_ATTENUATION_ADIOS)(void) {}

void FC_FUNC_(read_forward_arrays_adios,READ_FORWARD_ARRAYS_ADIOS)(void) {}

void FC_FUNC_(read_forward_arrays_undoatt_adios,READ_FORWARD_ARRAYS_UNDOATT_ADIOS)(void) {}

void FC_FUNC_(read_intermediate_forward_arrays_adios,READ_INTERMEDIATE_FORWARD_ARRAYS_ADIOS)(void) {}

void FC_FUNC_(read_mesh_databases_coupling_adios,READ_MESH_DATABASES_COUPLING_ADIOS)(void) {}

void FC_FUNC_(read_mesh_databases_mpi_cm_adios,READ_MESH_DATABASES_MPI_CM_ADIOS)(void) {}

void FC_FUNC_(read_mesh_databases_mpi_ic_adios,READ_MESH_DATABASES_MPI_IC_ADIOS)(void) {}

void FC_FUNC_(read_mesh_databases_mpi_oc_adios,READ_MESH_DATABASES_MPI_OC_ADIOS)(void) {}

void FC_FUNC_(read_mesh_databases_stacey_adios,READ_MESH_DATABASES_STACEY_ADIOS)(void) {}

void FC_FUNC_(save_forward_arrays_adios,SAVE_FORWARD_ARRAYS_ADIOS)(void) {}

void FC_FUNC_(save_forward_arrays_undoatt_adios,SAVE_FORWARD_ARRAYS_UNDOATT_ADIOS)(void) {}

void FC_FUNC_(save_intermediate_forward_arrays_adios,SAVE_INTERMEDIATE_FORWARD_ARRAYS_ADIOS)(void) {}

void FC_FUNC_(define_kernel_adios_variables,DEFINE_KERNEL_ADIOS_VARIABLES)(void) {}

void FC_FUNC_(write_kernels_cm_ani_adios,WRITE_KERNELS_CM_ADIOS)(void) {}

void FC_FUNC_(write_kernels_cm_iso_adios,WRITE_KERNELS_CM_ADIOS)(void) {}

void FC_FUNC_(write_kernels_oc_adios,WRITE_KERNELS_OC_ADIOS)(void) {}

void FC_FUNC_(write_kernels_ic_adios,WRITE_KERNELS_IC_ADIOS)(void) {}

void FC_FUNC_(write_kernels_boundary_kl_adios,WRITE_KERNELS_BOUNDARY_KL_ADIOS)(void) {}

void FC_FUNC_(write_kernels_source_derivatives_adios,WRITE_KERNELS_SOURCE_DERIVATIVES_ADIOS)(void) {}

void FC_FUNC_(write_kernels_hessian_adios,WRITE_KERNELS_HESSIAN_ADIOS)(void) {}

void FC_FUNC_(write_kernels_strength_noise_adios,WRITE_KERNELS_STRENGTH_NOISE_ADIOS)(void) {}
